### PR TITLE
docs: Release notes for 0.14.0.

### PR DESCRIPTION
**Pull request by determined-dsw**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/release-notes.txt [hunk 1]
```diff
@@ -4,6 +4,82 @@
...
+   <model-debug>`. The new guide will walk you step-by-step through
+   solving problems with a model in Determined, with a focus on testing
+   features incrementally until the model is fully working. It may also
+   be useful when porting new models to Determined.
+
+-  Documentation: Add a topic guide on :ref:`commands and shells
+   <commands-and-shells>`. It describes how to use Determined's support
+   for managing GPU-powered batch commands and interactive shells.
+
+-  **Breaking Change** Metrics: The trial and checkpoint API endpoints
```
Comment (justin-determined-ai) [inline docs/release-notes.txt:54]: non-blocking: move breaking changes to the top? also is this really about the REST API -- not sure if "metrics" count as a category...

File: docs/release-notes.txt [hunk 2]
```diff
@@ -4,6 +4,82 @@
...
+   can now return non-scalar metric values, which are represented as
+   JSON objects or `protobuf structs
```
Comment (justin-determined-ai) [inline docs/release-notes.txt:56]: non-blocking: Is the protobuf struct really a detail people need to know? They shouldn't see it if they interact with the public API (possibly I'm mistaken here)

File: docs/release-notes.txt [hunk 3]
```diff
@@ -4,6 +4,82 @@
...
+-  PyTorch: Upgrade the default PyTorch version to 1.7.0.
+
+-  TensorFlow: Upgrade the default TensorFlow versions to 1.15.5 and
+   2.4.1.
+
+-  REST API: Improve the performance of the experiments API.
+
+**Bug Fixes**
+
+-  API: Fix an issue where requesting checkpoint or trial details of a
```
Comment (justin-determined-ai) [inline docs/release-notes.txt:68]: blocking: "REST API"

File: docs/release-notes.txt [hunk 4]
```diff
@@ -4,6 +4,82 @@
...
+   trial that had non-scalar metric values associated with it would
+   fail.
+
+-  Database: Migrate ``public.trial_logs.id`` to be an ``int8`` in
+   Postgres, instead of an ``int4``. This avoids issues for customers
+   with extremely large amounts of trial logs. **Note**: This migration
+   may be more time-consuming than usual for deployments with large
```
Comment (justin-determined-ai) [inline docs/release-notes.txt:75]: blocking: Probably "will" be more time-consuming, right?

**Review by justin-determined-ai — APPROVED** [review]

File: docs/release-notes.txt (hunk 4 above)
Comment (determined-dsw) [reply docs/release-notes.txt:75]: I don't think we know that for sure in all of these cases, so I think "will" is a bit strong.

File: docs/release-notes.txt (hunk 2 above)
Comment (determined-dsw) [reply docs/release-notes.txt:56]: If the API is returning these values in this format, then the user will see this output.  I think it's helpful to have a description of the format of those outputs.

File: docs/release-notes.txt (hunk 3 above)
Comment (determined-dsw) [reply docs/release-notes.txt:68]: will be in next iteration

File: docs/release-notes.txt (hunk 1 above)
Comment (determined-dsw) [reply docs/release-notes.txt:54]: will be in next iteration

File: docs/release-notes.txt (hunk 4 above)
Comment (determined-dsw) [reply docs/release-notes.txt:75]: I changed my mind, will be in next iteration.

**Review by justin-determined-ai — APPROVED** [review]
Made some cosmetic fixes, LGTM

**Pull request merged**